### PR TITLE
Google Analytics Component

### PR DESCRIPTION
**Pull request by lolaodelola**
### Purpose
Create the Google Analytics component

**Google Analytics Disabled:**
screenshot 2018-11-01 at 07 44 38

**Google Analytics Enabled:**
screenshot 2018-11-01 at 07 44 51

**Customise Google Analytics triggered:**
screenshot 2018-11-01 at 07 45 06

### Notes
The text for the form labels are a little bigger than the designs and this is because there doesn't currently exist a font-size that's similar to the size in the design, however the styles for the general components (Card, Text, etc) are undergoing a refresh and so will look better once complete.
I've added a `workInProgress` flag for this component to hide it from view of the user until the whole Google Analytics feature is complete.

#### Staging Deployment

_This repo is CI/CD enabled and staging deployment should be available at:
https://<branch_name>.publish.buffer.com :smile:_

### CODE REVIEW

**Comment by karelm** [comment]
Hey @lolaodelola, I've tried out the branch but cant see the changes... I think this is likely because of the Work In Progress flag, could you remove that at all / set it to false so we can see the changes please?

File: packages/general-settings/components/GoogleAnalytics/index.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+              color={'black'}
+            >
+              Campaign Medium
+            </Text>
+            <Input
+              placeholder={'social'}
+            />
+          </div>
+          <div style={switchStyle}>
+            <Button>
```
Comment (karelm) [inline packages/general-settings/components/GoogleAnalytics/index.jsx]: @lolaodelola, you are probably already aware of this, but this button isn't triggering anything at present, so I don't think this is going to show / hide anything to users here until its all plumbed in. I guess the same goes with the text input's and their values, they arent going to go anywhere at present as their input data isnt being handled at the moment. But you probably already knew that 😀

Comment (lolaodelola) [reply packages/general-settings/components/GoogleAnalytics/index.jsx]: Thanks for this karel :) Yup, this is literally just the component, no real wiring yet, just displaying different sections when certain things the toggle & btn are triggered

File: packages/general-settings/components/GoogleAnalytics/index.jsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+      <div style={switchStyle}>
+        <Toggle
+          onText={'Campaign tracking:'}
+          offText={'Campaign tracking:'}
+          on={googleAnalyticsIsEnabled}
+          onClick={() => { onToggleGoogleAnalyticsClick(); }}
+        />
+      </div>
+    </div>
+    {showGACustomizationForm &&
```
[the remark this answers is not in the document]
Comment (lolaodelola) [reply packages/general-settings/components/GoogleAnalytics/index.jsx]: This section is only triggered once the button (l147 - l155) is clicked, if I use `googleAnalyticsEnabled` instead, we loose [this view](https://user-images.githubusercontent.com/7262392/47839640-56310780-ddab-11e8-8d70-7d8c78c25ad7.png) & instead have just have [this view](https://user-images.githubusercontent.com/7262392/47839641-56310780-ddab-11e8-9efa-dfe9a4126214.png).

**Comment by lolaodelola** [comment]
Hey @karelm I've toggled the `workInProgress` flag if you wanna check that out, I'll change it back once everyone is done reviewing.

File: packages/general-settings/components/GoogleAnalytics/index.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (karelm) [reply packages/general-settings/components/GoogleAnalytics/index.jsx]: awesome, just wondered 👍 thanks for digging in there 😃

File: packages/general-settings/reducer.js [hunk 3]
```diff
@@ -52,9 +66,13 @@ export const actions = {
     type: actionTypes.SET_DIRECT_POSTING,
     profileId: action.profileId,
   }),
-  handleOnSelectLinkShortenerChange: ({ profileId, domain }) => ({
-    type: actionTypes.CHANGE_SELECTED_LINK_SHORTENER,
-    profileId,
-    domain,
```
Comment (karelm) [inline packages/general-settings/reducer.js]: I think you may have accidentally deleted this function here? could you take a look at all please?

File: packages/general-settings/index.js [hunk 4]
```diff
@@ -10,18 +10,25 @@ export default connect(
...
       onSetUpDirectPostingClick: () => {
         dispatch(actions.handleSetUpDirectPostingClick({
           profileId: ownProps.profileId,
         }));
       },
-      onLinkShortenerOptionSelect: (event) => {
-        dispatch(actions.handleOnSelectLinkShortenerChange({
-          profileId: ownProps.profileId,
-          domain: event.target.value,
-        }));
```
Comment (karelm) [inline packages/general-settings/index.js]: I think you may have accidentally deleted a function here @lolaodelola, could you take a look for me please?

File: packages/general-settings/reducer.js (hunk 3 above)
Comment (lolaodelola) [reply packages/general-settings/reducer.js]: Ahh apologies!

**Comment by karelm** [comment]
Hey @lolaodelola, 
I'm happy with the work here on this PR, but I'm not exactly sure about getting this out on live at present. It doesn't seem like its complete, or ready to get out at the moment, so probably could do with some more time finishing the piece of work in its entirity. 
I think it looks like you are kind of going for a feature flag approach with your code, but following on from the discussion the other week in the sync session (question asked by Ana around long lived branches https://paper.dropbox.com/doc/Publish-Teddy-Weekly-Sync--AQ1s1L6iW_h_p8XzLDn3H2caAg-eJsqiQ4jt8MoCeFYSnF5G#:uid=431719844707089440675858&h2=Friday%2C-October-26%2C-2018) I think it was stated that features are something belonging to a users account plan, and not relating to how we manage showing / hiding code for our end users (from a developers perspective). 
This isn't a major issue, but I think it would be great to keep this branch open and complete the deliverables of this issue in its totality before merging it in. I know you don't want to have the branch hanging around forever, but I'm not sure that the message was quite clear in that call, so thought I'd mention it here for reference.
Great work otherwise, keep on doing what you are doing! 💯

**Comment by lolaodelola** [comment]
Great, thanks @karelm, I'll leave this open until the other work is done :)

**Comment by lolaodelola** [comment]
Closing this as it looks like these changes have been merged into master here: https://github.com/bufferapp/buffer-publish/pull/419 by @MayaUribe

**Pull request closed without merging**